client/me/purchases/manage-purchase/index.tsx

```diff
@@ -638,6 +639,49 @@ class ManagePurchase extends Component<
 		return null;
 	}
 
+	renderRefundText() {
+		const { purchase, translate } = this.props;
+
+		// Hide if refund window has lapsed.
+		if ( ! hasAmountAvailableToRefund( purchase ) || ! purchase?.mostRecentRenewDate ) {
+			return;
+		}
+
+		const ONE_DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000;
+		const lastRenewalDate = new Date( purchase.mostRecentRenewDate );
+		const refundPeriodEndDate = new Date(
+			lastRenewalDate.getTime() + purchase.refundPeriodInDays * ONE_DAY_IN_MILLISECONDS
+		);
+
+		// We use relative timing when the subscription expires that day for added clarity.
+		const today = new Date();
+		refundPeriodEndDate.setHours( 0, 0, 0, 0 );
+
+		today.setHours( 0, 0, 0, 0 );
+		refundPeriodEndDate.setHours( 0, 0, 0, 0 );
+
+		let refundText;
+
+		if ( today.getTime() === refundPeriodEndDate.getTime() ) {
+			refundText = translate( 'Refund window ends today', {
+				args: {
+					refundAmount: purchase.refundText,
+				},
+			} );
+		} else if ( today < refundPeriodEndDate ) {
+			refundText = translate( 'Refund available until {{refundDate/}}', {
+				components: {
+					refundDate: <FormattedDate date={ refundPeriodEndDate } format="LL" />,
+				},
+				args: {
+					refundAmount: purchase.refundText,
+				},
+			} );
+		}
+
```

Comment:
Should there be an early return here if `refundText` is still empty?

In practice that shouldn't happen, since if the subscription isn't refundable it would have already returned early above.  But since `refundText` is being defined in an `if/elseif` then it's at least theoretically possible that it never gets set to anything, so it might be worth protecting against.